ext/node/polyfills/_http_outgoing.ts

```diff
@@ -491,19 +493,53 @@ Object.defineProperties(
       return ret;
     },
 
+    /** Right after socket is ready, we need to writeHeader() to setup the request and
+     *  client. This is invoked by onSocket(). */
+    _flushHeaders() {
+      if (!this._headerSent) {
+        this._headerSent = true;
+        this._writeHeader();
+      }
+    },
+
     // deno-lint-ignore no-explicit-any
     _send(data: any, encoding?: string | null, callback?: () => void) {
-      if (!this._headerSent && this._header !== null) {
-        this._writeHeader();
-        this._headerSent = true;
+      // if socket is ready, write the data after headers are written.
+      // if socket is not ready, buffer data in outputbuffer.
+      if (
+        this.socket && !this.socket.connecting && this.outputData.length === 0
```

Comment:
Does this handle situation when the socket is destroyed when this method is called?

Reply:
I don't think this part handles destroyed state. I think destroyed state is naturally ignored by non-functioning `_bodyWriter` which is used inside `_writeRaw()` function.